Luke7787/ImmaculateInventors
# Feature/s3 storage

Summary:
Made it so that images get stored in the AWS S3 database.

Testing:
1. Pull code
2. Go into the terminal into the express backend folder
3. `npm install`
4. `npm run dev`
5. Have a jpeg ready
6. go into boomerang
7. POST: http://localhost:8000/upload
8. Multipart (instead of JSON)
9. click file instead of text on the right side
10. For the name, put "imageFile"
11. for the file, just click on it and select an image from your computer.

____________________________________________________________________________________

After you send the image, log into AWS:

1. Go to: https://signin.aws.amazon.com/signin?redirect_uri=https%3A%2F%2Fconsole.aws.amazon.com%2Fconsole%2Fhome%3FhashArgs%3D%2523%26isauthcode%3Dtrue%26nc2%3Dh_ct%26src%3Dheader-signin%26state%3DhashArgsFromTB_us-east-2_8ef9549c7387bfe7&client_id=arn%3Aaws%3Asignin%3A%3A%3Aconsole%2Fcanvas&forceMobileApp=0&code_challenge=q1dvsoo20OfcU2WT4AI-UMSwd71ysJu8HLo4wIFvNF8&code_challenge_method=SHA-256

2. click "IAM user" and enter the following credentials:
          ID: 058264393406
          Username: Inventors
          Password: ImmaculateInventors! 


3. In the search bar at the top, search for "S3" and click the first service that pops up.

4. Click on the immaculateitemimages bucket and you should be able to see your image. 

5. In order to check that it works click on the image you uploaded (check the date/time it says it was uploaded) and click open to see if it's the same.
